com.unity.shadergraph/Editor/GraphUI/DataModel/Searcher/ShaderGraphSearcherDatabaseProvider.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+using UnityEditor.GraphToolsFoundation.Overdrive;
+using UnityEditor.GraphToolsFoundation.Searcher;
+using UnityEngine.GraphToolsFoundation.Overdrive;
+
+namespace UnityEditor.ShaderGraph.GraphUI.DataModel
+{
+    public class ShaderGraphSearcherDatabaseProvider : DefaultSearcherDatabaseProvider
+    {
+        public ShaderGraphSearcherDatabaseProvider(Stencil stencil)
+            : base(stencil)
+        {
+        }
+
+        SearcherDatabaseBase CreateNodeDatabaseFromRegistry(IGraphModel graphModel)
+        {
+            // TODO: Get all keys and names from registry. Make categories by nesting SearcherItems.
+            return new SearcherDatabase(new List<SearcherItem>
+            {
+                new RegistryNodeSearcherItem(graphModel, new PlaceholderRegistryKey(), "TODO")
+            });
+        }
+
+        public override List<SearcherDatabaseBase> GetGraphElementsSearcherDatabases(IGraphModel graphModel)
+        {
+            var databases = base.GetGraphElementsSearcherDatabases(graphModel);
+            databases.Add(CreateNodeDatabaseFromRegistry(graphModel));
+            return databases;
+        }
+
+        SearcherDatabaseBase CreateTypeDatabaseFromRegistry()
```

Comment:
If all of our TypeHandle stuff could go through the same place that'd be great. Since we're mostly focused on appearance and connectivity on our end, that should spare us some of the painful details. Let's look over this once we get to types -- I'm hoping we'll find a convenient registry type <-> TypeHandle mechanism.